### ISSUE TITLE
Setup newsletter

### ISSUE THREAD

**Issue by francisco-core** [PrivacyLx/privacylx-issue-tracker#6]
We need to evaluate our options of how we are going to let people know that our events are going to happen.
Some options discussed were:
* mailinglist
* newsletter
* meetup.com

**Comment by anadahz** [comment]
I have renamed this ticket to better describe what we want to do (upon discussion with @francisco-core).

Adding some projects and examples:
- [Tor Newsletter](https://newsletter.torproject.org/) is using [PHPMailer](https://github.com/PHPMailer/PHPMailer).
- [Mailtrain](https://mailtrain.org/), seems to the be the most supported and up to date self-hosted newsletter tool.

Please feel free to add your suggestions.

**Comment by francisco-core** [comment]
I found an ansible role and playbook for maltrain https://github.com/vaizard/mage-mailtrain
but I don't know if it is compatible with the dev branch of mailtrain, which seems to be the most updated one: https://github.com/Mailtrain-org/mailtrain/tree/development

**Comment by francisco-core** [comment]
It's seems is is going to be quite a headache. Before we dive deep into this. I would suggest considering the alternative of using a [riseup mailing-list](https://riseup.net/en/lists/list-admin/configuration/creating-lists). We need no configuration on our end and we can always migrate elsewhere later.

This is what the people of [cryptorave](https://cryptorave.org/)

For people to subscribe, we'll have to point people to something like this: https://lists.riseup.net/www/subscribe/cryptorave-boletim

**Comment by francisco-core** [comment]
added ["subscribe newsletter" branch](https://github.com/PrivacyLx/website/tree/newsletter) to website code as a preparation for this issue resolution. Here's how it currently looks:

![privacy-lx-newsletter](https://user-images.githubusercontent.com/32313429/55658829-9da24080-57ee-11e9-8c94-b448868e5db5.png)

**Comment by francisco-core** [comment]
I've signed up for riseup's newsletter service. They should reply in the next couple of days on the approval of a newsletter there for our organization

**Comment by anadahz** [comment]
> I've signed up for riseup's newsletter service. 

RiseUp "newsletter service" is actually a mailing list.
We should check if it's possible to host a newsletter service with Discourse.

**Comment by francisco-core** [comment]
I tried out the idea @anadahz suggested in https://github.com/PrivacyLx/privacylx-issue-tracker/issues/19 about doing this with the [custom wizard plugin](https://meta.discourse.org/t/custom-wizard-plugin/73345/) but I won't work for our use-case because as of today it still doesn't support anonymous visitors to submit forms. Which means that if the anonymous visitor shows up there they will be prompted with a login/signup page which defeats the whole purpose of doing a quick "enter your email to subscribe to the newsletter".

Related comment: (https://meta.discourse.org/t/custom-wizard-plugin/73345/315)
>> One more question, is any way to create a survey for anonymous visitors?
>
> Not yet, but soon I think.

**Comment by francisco-core** [comment]
I think we should move on with the PHP discourse newsletter API idea

The mailchimp signup forms do have some protection against bots

```html
<!-- Begin Mailchimp Signup Form -->
<link href="//cdn-images.mailchimp.com/embedcode/horizontal-slim-10_7.css" rel="stylesheet" type="text/css">
<style type="text/css">
	#mc_embed_signup{background:#fff; clear:left; font:14px Helvetica,Arial,sans-serif; width:100%;}
	/* Add your own Mailchimp form style overrides in your site stylesheet or in this style block.
	   We recommend moving this block and the preceding CSS link to the HEAD of your HTML file. */
</style>
<style type="text/css">
	#mc-embedded-subscribe-form input[type=checkbox]{display: inline; width: auto;margin-right: 10px;}
	#mergeRow-gdpr {margin-top: 20px;}
	#mergeRow-gdpr fieldset label {font-weight: normal;}
	#mc-embedded-subscribe-form .mc_fieldset{border:none;min-height: 0px;padding-bottom:0px;}
</style>
<div id="mc_embed_signup">
<form action="https://codastory.us12.list-manage.com/subscribe/post?u=2917466ad5ae7d0be32196119&amp;id=610841f0e2" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate="">
    <div id="mc_embed_signup_scroll">
	<label for="mce-EMAIL">Sign up for our upcoming Coda Currents newsletter</label>
	<input type="email" value="" name="EMAIL" class="email" id="mce-EMAIL" placeholder="email address" required="">
    <!-- real people should not fill this in and expect good things - do not remove this or risk form bot signups-->
    <div style="position: absolute; left: -5000px;" aria-hidden="true"><input type="text" name="b_2917466ad5ae7d0be32196119_610841f0e2" tabindex="-1" value=""></div>
    <div class="clear"><input type="submit" value="Subscribe" name="subscribe" id="mc-embedded-subscribe" class="button"></div>
    </div>
</form>
</div>

<!--End mc_embed_signup-->
```

Notice there the comment and the `position: absolute; left: -5000px;` then what we have to do is to see if that is filled with anything other than `""` and drop the request if it is

If this is enough to fight most spam, then it should be enough for us @anadahz

**Comment by anadahz** [comment]
@francisco-core sound OK for me. Do you have an example website to check?
Just to clear things in case of confusion this is just an example and we are not going to use Mailchimp but the Discourse forum for newsletters. ;)

**Comment by francisco-core** [comment]
> @francisco-core sound OK for me. Do you have an example website to check?
> Just to clear things in case of confusion this is just an example and we are not going to use Mailchimp but the Discourse forum for newsletters. ;)

Yeah. It's in the middle of this article: https://codastory.com/authoritarian-tech/surveillance/uyghurs-turkey-whatsapp-wechat/

No, no mailchimp stuff for us. But we can borrow their code for the signup safety

**Comment by francisco-core** [comment]
I am working on this and on the php script now @anadahz
